src/js/player.js

```diff
@@ -1213,6 +1342,16 @@ class Player extends Component {
    * @private
    */
   handleTechSourceset_(event) {
+    this.updateSourceCaches_(event.src);
```

Comment:
my main concern here is that we accidentally set the source cache to the wrong value.
Specifically, around behavior like https://github.com/videojs/video.js/blob/master/test/unit/sourceset.test.js#L534-L601 where we end up with the first source triggering the second sourceset

Maybe we just need more unit/integration tests specifically around this?

Reply:
Yeah I have a lot of unit test changes, I just don't want to port them over until the other big pr is in: #5038